### ISSUE TITLE
squinktronix-plug1

### ISSUE THREAD

**Issue by squinkylabs** [VCVRack/community#779]
I'm planning on putting these into the library, so this is just a "heads up".

The repo is [here](https://github.com/squinkylabs/SqHarmony), it's structure pretty much like the Squinky Labs repo is. should be easy to build. I build them on the toolchain np.

One wrinkle - this repo has no license. Which is intentional, but might present issues for redistributing in the library. Let  me know if I should change anything.

thanks!

**Comment by AndrewBelt** [comment]
I have some suggestions:
- Module slugs are specific to the plugin slug, so there's no need for the "sqh-" prefix.
- The plugin slug contains "-plug1" which is redundant because all plugin slugs refer to plugins.

**Comment by cschol** [comment]
Missing license key in plugin manifest: https://github.com/squinkylabs/SqHarmony/issues/29

**Comment by squinkylabs** [comment]
@AndrewBelt yeah, you are totally right about the redundant stuff in the slug. I guess it's ancient bad habit. Don't really want to change it as it will make life more difficult for the handful of people who have it already. Although I guess they would just end up with two copies.

**Comment by squinkylabs** [comment]
@cschol yes, I mentioned that there is no license file when I opened this issue. Could you advise me on what I should do? I guess I can create a custom license. Do you have any examples I could look at?

**Comment by cschol** [comment]
Pick one: https://spdx.org/licenses/

**Comment by squinkylabs** [comment]
I'm sorry for being dense. I think you are providing those as possible examples that I could use/adapt? I think my conclusion is that VCV library requires some kind of license text that can be linked to, but it doesn't matter what it says?

As I say, still slightly confused. The build toolchain was happy to build on all four (or three) platforms from my repo, but I guess there is also a requirement not enforced they the build toolchain that you must also have license?

I for sure would like to resolve this so that my modules may be downloaded for free from your library.

Are the requirements documented somewhere?

Thanks!

**Comment by cschol** [comment]
The manifest and its required fields are documented here (linking to specifically the license field): https://vcvrack.com/manual/Manifest#license

**Comment by squinkylabs** [comment]
Thanks, that's super useful. Minor note: all the links on the page are dead. They link to eula pages in vcv.com that do not exist.

**Comment by squinkylabs** [comment]
ok, I made a nice license. If you could put current main, aa8c342c10e0e53ea66140d88bc001d259a45d10 into the library that would be great!

**Comment by cschol** [comment]
Your plugin has been integrated and will be available for all supported platforms with the next Plugin Manager update. Please let us know in **this** issue when a new version is available. We will reopen it. Thank you for providing your plugin to the Rack community.

**Comment by squinkylabs** [comment]
It seems like the plugin is not actually in the library, although all the other ones are, so presumably there has been a plugin manager update?

**Comment by AndrewBelt** [comment]
One of the module objects in the plugin manifest doesn't include a `tags` property, so I had to make the website handle this case. Your plugin now appears at https://library.vcvrack.com/squinktronix-plug1

**Comment by squinkylabs** [comment]
hey, I could use a new build!
branch: main
sha: 3c17b016daf3af9fa45571a3183ffcdfa8a6598c
version 2.2.4

tx!

**Comment by squinkylabs** [comment]
Ok, got a new module and a bug fix or two.

SHA: 6f29b84e18923b6c49cf0f59e14daf35fe2790ad
repo: https://github.com/squinkylabs/SqHarmony
branch: main
version: 2.2.4

**Comment by squinkylabs** [comment]
I got a new module.

branch: main
sha: e4c51ec78222b73ff81ecd6f2c9bce006a006a99

**Comment by cschol** [comment]
Can you doublecheck that SHA? It doesn't seem to exist in the repo.

**Comment by squinkylabs** [comment]
Wow - don't know what happened there.
This sha should work: fa308c0b704f3d16f6be3f1871ed8500c9f6d755

**Comment by squinkylabs** [comment]
Got a bug fix version ready.
version 2.2.6
branch: main
sha: 64d0b2f71e66e83adbe3d6dd10a913e27aca1536

**Comment by squinkylabs** [comment]
I need a new build to make this look right in 2.7.x

on my main branch
sha: 53b3e5da632926415c8370adab7fe9c6038438b9

**Comment by AndrewBelt** [comment]
The last build fails to load on Windows due to a bug with our toolchain, so I am reverting this release on the VCV Library until we fix https://github.com/VCVRack/rack-plugin-toolchain/issues/41.
